include/simics/dmllib.h

```diff
@@ -119,6 +124,20 @@ static inline int64 DML_mod(int64 a, int64 b, const char *filename, int lineno)
         return a % b;
 }
 
+static inline double DML_divf(double a, double b)
+{
+    if (unlikely(b == 0)) {
+        if (unlikely(isnan(a))) {
+            return a * b;
+        } else if (unlikely(a == 0)) {
+            return copysign(NAN, a * b);
+        } else {
+            return copysign(INFINITY, a * b);
+        }
+    }
+    return a / b;
+}
```

Comment:
Everything this does is horribly redundant in the extremely common (read: _only_) case where IEEE 754 is supported.
This could justify:
```C
#if defined(__COVERITY__) || !defined(__STDC_IEC_559__)
static inline double DML_divf(double a, double b)
{
    ...
}
#else
#define DML_divf(a, b) ((a)/(b))
#endif
```
Opinions?